### PR TITLE
Support for snmpv3 context

### LINKED ISSUE #17: Add support for snmpv3 contexts
I would like to use snmpv3 contexts in my test and the current version of this library does not make it possible. The pysnmp library seems to have support for contexts.

### PR DESCRIPTION
**Pull request by ecklm**
Closes #17

### CODE REVIEW

**Comment by ecklm** [comment]
Closes #17

File: src/SnmpLibrary/library.py [hunk 1]
```diff
@@ -97,7 +99,8 @@ def open_snmp_v3_connection(self, host, user, password='',
                                 encryption_password=None,
                                 authentication_protocol=None,
                                 encryption_protocol=None, port=161,
-                                timeout=1.0, retries=5, alias=None):
+                                timeout=1.0, retries=5, alias=None,
+                                context_name=null):
```
Comment (hthiery) [inline src/SnmpLibrary/library.py:103]: could you please add the description into the documentation section below

File: src/SnmpLibrary/library.py [hunk 2]
```diff
@@ -249,7 +252,8 @@ def _get(self, oid, idx=(0,), expect_display_string=False):
             self._active_connection.cmd_gen.getCmd(
                 self._active_connection.authentication_data,
                 self._active_connection.transport_target,
-                oid
+                oid,
+                contextName=self._active_connection.context_name,
```
Comment (hthiery) [inline src/SnmpLibrary/library.py:256]: please remove the ',' from the end of the line

File: src/SnmpLibrary/library.py [hunk 3]
```diff
@@ -307,7 +311,8 @@ def _set(self, *oid_values):
             self._active_connection.cmd_gen.setCmd(
                 self._active_connection.authentication_data,
                 self._active_connection.transport_target,
-                *oid_values
+                *oid_values,
+                contextName=self._active_connection.context_name,
```
Comment (hthiery) [inline src/SnmpLibrary/library.py:315]: please remove the ',' from the end of the line

File: src/SnmpLibrary/library.py [hunk 4]
```diff
@@ -390,7 +395,8 @@ def walk(self, oid):
             self._active_connection.cmd_gen.nextCmd(
                 self._active_connection.authentication_data,
                 self._active_connection.transport_target,
-                oid
+                oid,
+                contextName=self._active_connection.context_name,
```
Comment (hthiery) [inline src/SnmpLibrary/library.py:399]: please remove the ',' from the end of the line

**Comment by hthiery** [comment]
could you please also add a signed-off-by on your commits?

File: src/SnmpLibrary/library.py (hunk 1 above)
Comment (ecklm) [reply src/SnmpLibrary/library.py:103]: Done in 0f18d150a6ac21aaf7dcf4dfddbf732bcacc3ace

File: src/SnmpLibrary/library.py (hunk 2 above)
Comment (ecklm) [reply src/SnmpLibrary/library.py]: Done in 0f18d150a6ac21aaf7dcf4dfddbf732bcacc3ace

File: src/SnmpLibrary/library.py (hunk 3 above)
Comment (ecklm) [reply src/SnmpLibrary/library.py]: Done in 0f18d150a6ac21aaf7dcf4dfddbf732bcacc3ace

File: src/SnmpLibrary/library.py (hunk 4 above)
Comment (ecklm) [reply src/SnmpLibrary/library.py]: Done in 0f18d150a6ac21aaf7dcf4dfddbf732bcacc3ace

**Comment by ecklm** [comment]
> could you please also add a signed-off-by on your commit messages?

I did it for both commits.

**Comment by hthiery** [comment]
thank you

**Comment by ecklm** [comment]
Is there a plan to add this to a new release so it's available via pypi?

**Comment by hthiery** [comment]
I will try to do this in the next days

**Comment by number492** [comment]
Hi! Do you have an estimate for when the new release can happen? We plan to drop our fork when it becomes available.

**Comment by ecklm** [comment]
Is there any plan making a new release soon?

**Pull request merged**